webapp/src/app/shared/map-loader.service.ts

```diff
@@ -27,11 +27,7 @@ export class MapLoaderService {
 		reader.onload = (e: any) => {
 			try {
 				const map = JSON.parse(e.target.result);
-				if (!map.mapHeight) {
-					throw new Error('invalid map');
-				}
-				map.filename = file.name;
-				this._map.next(map);
+				this.loadRawMap(map);
```

Comment:
`map.filename` should be set to `file.name`. It's needed to save the changed file with the same name again